### PR TITLE
allow breakpoints to have associated commands

### LINKED ISSUE #115: Feature Request: Ignoring Breakpoints
It would be cool if there was a way to ignore a breakpoint n times, similar to GDB's ignore command. This would really help with debugging nested loops with several iterations. 

It seems doable by extending https://github.com/insou22/mipsy/blob/main/crates/mipsy_lib/src/compile/mod.rs#L38 to a HashMap of addresses and remaining hit counts and modifying this condition: https://github.com/insou22/mipsy/blob/main/crates/mipsy_interactive/src/interactive/mod.rs#L409 to take into account the remaining hit counts?

### PR DESCRIPTION
**Pull request by joooooooooooooooooooooooooooooooooooosh**
This also introduces the `breakpoint temporary` command, which creates a breakpoint that deletes itself after being hit
The ability to ignore breakpoints for a specified number of times has also been added
fixes #115

### CODE REVIEW

**Comment by Dylan-Brotherston** [comment]
Can we have a count?
Delete after N times hit.

**Comment by joooooooooooooooooooooooooooooooooooosh** [comment]
> Can we have a count? Delete after N times hit.

Yeah I just remembered that issue after opening this PR, I can add it pretty easily

File: crates/mipsy_interactive/src/interactive/commands/breakpoint.rs [hunk 1]
```diff
@@ -79,14 +87,16 @@ fn get_long_help() -> String {
         "help breakpoint".bold(),
         "<subcommand>".purple().bold(),
         "<subcommand>".purple(),
+        "temporary".purple(),
+        "ignore".purple(),
     )
 }
 
-fn breakpoint_insert(state: &mut State, label: &str, args: &[String], remove: bool) -> Result<String, CommandError> {
+fn breakpoint_insert(state: &mut State, label: &str, mut args: &[String], remove: bool) -> Result<String, CommandError> {
```
Comment (Dylan-Brotherston) [inline crates/mipsy_interactive/src/interactive/commands/breakpoint.rs:95]: Is a temporary breakpoint created with
`breakpoint insert temporary <address>`
or
`breakpoint temporary <address>`
This help says to do both?
`breakpoint temporary temporary <address>`

File: crates/mipsy_interactive/src/interactive/commands/commands.rs [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+use rustyline::error::ReadlineError;
+
+use crate::{interactive::{editor, error::CommandError}, prompt};
+
+use super::*;
+
+pub(crate) fn commands_command() -> Command {
+    command(
+        "commands",
+        vec![],
```
Comment (Dylan-Brotherston) [inline crates/mipsy_interactive/src/interactive/commands/commands.rs:12]: Add some alternatives
`command`, `com`, `comms`

**Review by Dylan-Brotherston — CHANGES_REQUESTED** [review]

File: crates/mipsy_interactive/src/interactive/commands/breakpoint.rs (hunk 1 above)
Comment (joooooooooooooooooooooooooooooooooooosh) [reply crates/mipsy_interactive/src/interactive/commands/breakpoint.rs:95]: all three are technically valid, I allowed `breakpoint temporary` as a shortcut for `breakpoint insert temporary` and due to quirks in the parsing logic `breakpoint temporary temporary` is valid syntax for creating a temporary breakpoint
I'll try and remove the validity of `breakpoint temporary temporary`

Comment (Dylan-Brotherston) [reply crates/mipsy_interactive/src/interactive/commands/breakpoint.rs:95]: If `breakpoint temporary` is valid. I don't see the need for `breakpoint insert temporary`.

But the main issue is that the help text makes it sound like you must use `breakpoint temporary temporary`

Comment (joooooooooooooooooooooooooooooooooooosh) [reply crates/mipsy_interactive/src/interactive/commands/breakpoint.rs]: Yeah I decided it was neatest to only allow `breakpoint temporary` and updated help text to reflect this

**Pull request merged**